src/Processors/Formats/Impl/CHColumnToArrowColumn.cpp

```diff
@@ -356,6 +357,18 @@ namespace DB
         }
     }
 
+    static void checkIfIndexesTypeIsExceeded(const std::shared_ptr<arrow::DataType> & arrow_type, size_t dict_size)
+    {
+        const auto & dict_indexes_arrow_type = assert_cast<const arrow::DictionaryType *>(arrow_type.get())->index_type();
+        /// We use UInt32 or UInt64 type for indexes. It makes sense to check overflow only for UInt32.
+        const auto * indexes_uint32_type = typeid_cast<const arrow::UInt32Type *>(dict_indexes_arrow_type.get());
```

Comment:
Alternatively we can compare `dict_indexes_arrow_type->bit_width()` and `std::bit_width(dict_size)`, but I don't think it's worth changing. The only reason is that if we change `getArrowTypeForLowCardinalityIndexes` one more time (e.g. to use uint16)  we will need not to forget this place.